src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/FrameHeaders.cs

```diff
@@ -15,9 +15,10 @@ public abstract class FrameHeaders : IHeaderDictionary
     {
         protected bool _isReadOnly;
         protected Dictionary<string, StringValues> MaybeUnknown;
-
         protected Dictionary<string, StringValues> Unknown => MaybeUnknown ?? (MaybeUnknown = new Dictionary<string, StringValues>(StringComparer.OrdinalIgnoreCase));
 
+        public long? ContentLength { get; set; }
```

Comment:
With throw an InvalidOp for both Request and Response as will need to be manually set rather than input Request set (so server error 500)